crates/goose/src/session/session_manager.rs

```diff
@@ -1255,6 +1267,11 @@ impl SessionStorage {
         self.get_session(&new_session.id, true).await
     }
 
+    async fn fork_session(&self, session_id: &str) -> Result<Session> {
```

Comment:
so fork session is the same as copy session, except for that it also copies the name? not sure we need that